Source/Programs/CLI.h

```diff
@@ -36,7 +40,19 @@ inline bool loadTextFile(const char* filename,WebAssemblyText::File& outFile)
 	{
 		// Print any parse errors;
 		std::cerr << "Error parsing WebAssembly text file:" << std::endl;
-		for(auto error : outFile.errors) { std::cerr << error->message.c_str() << std::endl; }
+
+		std::stringstream ss(wastString);
+		std::string to;
+		uint32 currentLineNumber = 0;
+
+		std::sort(outFile.errors.begin(), outFile.errors.end(), sortByLines);
+		for(auto error : outFile.errors)
+		{
+			std::cerr << filename << ":" << error->message.c_str() << std::endl;
+			while(currentLineNumber != error->line && std::getline(ss,to,'\n')){ currentLineNumber++; }
+			std::cerr << to << std::endl;
+			std::cerr << std::setw(error->column) << "^" << std::endl;
+		}
```

Comment:
I'd prefer this to be a little less clever. I'd split this into two loops: one that steps through wastString and adds the index of each newline to an array, and one that prints each error, using the newline index to find the slice of the string corresponding to that line.